src/xrpld/app/tx/detail/Escrow.cpp

```diff
@@ -363,13 +367,34 @@ EscrowFinish::calculateBaseFee(ReadView const& view, STTx const& tx)
     return Transactor::calculateBaseFee(view, tx) + extraFee;
 }
 
+TER
+EscrowFinish::preclaim(PreclaimContext const& ctx)
+{
+    auto const kEscrow =
+        keylet::escrow(ctx.tx[sfOwner], ctx.tx[sfOfferSequence]);
+    auto const sleEscrow = ctx.view.read(kEscrow);
+    if (!sleEscrow)
+        return tecNO_TARGET;
+
+    // NOTE: Escrow payments cannot be used to fund accounts.
+    AccountID const dst = sleEscrow->getAccountID(sfDestination);
+    auto const sleDst = ctx.view.read(keylet::account(dst));
+    if (!sleDst)
+        return tecNO_DST;
+
+    if (auto const err =
+            credentials::valid(ctx, ctx.tx[sfAccount], dst, sleDst);
+        !isTesSuccess(err))
+        return err;
+
+    return tesSUCCESS;
+}
+
 TER
 EscrowFinish::doApply()
 {
     auto const k = keylet::escrow(ctx_.tx[sfOwner], ctx_.tx[sfOfferSequence]);
     auto const slep = ctx_.view().peek(k);
-    if (!slep)
```

Comment:
IMO this if should stay with a `tecINTERNAL` error.

Reply:
fixed